src/jpgimage.cpp

```diff
@@ -84,6 +84,14 @@ bool markerHasLength(byte marker) {
          marker == dqt_ || marker == dri_ || marker == com_ || marker == sos_;
 }
 
+void readSegmentSize(const byte marker, BasicIo& io, std::array<byte, 2>& buf, std::uint16_t& size) {
```

Comment:
Modern C++ theory says to avoid non const references and use return values instead. In this case, you could return an std::pair of std::array and std::uint16_t. Makes sense here too. The values here are unused before this function.

As for Io, that should probably remain a pointer.

Reply:
Yes, I thought about using returning a `std::pair` in a 2nd iteration and then I forgot. It will probably simplify the code in the scopes where I am calling this function. I will follow this suggestion 👍 

Regarding the `io` param, I tend to use a reference when I am sure that the object cannot be `nullptr` and this prevents me for the need to check for it. In this case, since it is a private function in this translation unit I could also leave it as a pointer and assume that the pointer will not be `nullptr` ... but I would argue that there is no benefit of doing so.